### ISSUE TITLE
Unable to share saved SQL queries - user gets an error wen trying to load the query

### ISSUE THREAD

**Issue by yousoph** [apache/superset#31367]
## Description
Users are unable to share saved SQL queries to other users. The owner of a saved query can access it via the shared URL, but any other user attempting to access the same URL receives an error message: “The query couldn’t be loaded."

## How to reproduce the bug
1. Log in and save a query in SQL Lab.
2. Copy the query URL using the "Copy Query URL" option.
3. Share the URL with a different user.
4. Have the other user try to load the shared query URL.
5. Observe that the second user receives the error message: “The query couldn’t be loaded.”

## Expected results
Any user with access to SQL Lab should be able to load and view saved SQL queries via a shared URL.

## Actual results
Only the creator of the query can load the query from the shared URL. Other users receive an error message stating: "The query couldn’t be loaded."

### Superset version

master / latest-dev

### Python version

3.9

### Node version

16

### Browser

Chrome

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [ ] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by geido** [comment]
---

### 🎉 **Preset Bounty Available: $150 USD** 🎉

To claim this bounty, please carefully follow the steps below.

---

#### 📋 **Steps to Participate**

1. **Review Guidelines:**  
   Read through the [Preset Bounty Program Contribution Guide](https://preset.notion.site/preset-bounty-program-apache-superset) for complete details on bounty requirements.

2. **Show Your Interest:**  
   Complete the [Preset Bounty Program Survey](https://docs.google.com/forms/d/e/1FAIpQLSeDsR1iWvPxFydUIXs2agsUg3kMDaAooeD4P9P1L5sI0Yu5QQ/viewform?usp=sharing) and comment this issue to express your interest.

3. **Join the Slack Channel:**  
   After completing the survey, you’ll receive an invitation to the dedicated [Apache Superset Slack](https://join.slack.com/t/apache-superset/shared_invite/zt-2be0drwz8-bxPfkdz28ozzk1Iox29ufg) channel.

4. **Get Assigned:**  
   To officially start, ensure a **Bounty Program Manager** has assigned you to this issue.

5. **Submit Your Solution:**  
   When ready, submit your solution with the `Fixes #{issue_number}` notation in your Pull Request description.

6. **Claim Your Bounty:**  
   Sign up at [GitPay.me](https://gitpay.me/) and submit your solution via: [https://gitpay.me/#/task/1013](https://gitpay.me/#/task/1013)

---

#### 💡 **Additional Notes**

- Only developers assigned by a **Bounty Program Manager** should start working on this issue to win the bounty.
- Be sure to follow the guide closely to avoid any delays in payment. Please, allow a few days after your PR has been merged for the bounty to be released.

Good luck, and happy coding! 🎉

**Comment by LevisNgigi** [comment]
Hey @geido I would love to work on this.

**Comment by lethehoa** [comment]
Still got the problem **The query couldn't be loaded** in version 4.1.1

**Comment by LevisNgigi** [comment]
> Got the problem **The query couldn't be loaded** in version 4.1.1, but I can't upgrade to the newest version now. Is there any way to fix this without upgrade?

Hey @lethehoa the fix is in 5.0.0 but I could help fix that for you if you do not want to upgrade.

**Comment by anjosma** [comment]
Any plans to include this fix on v4.1.3?

**Comment by SandadiLokanathaReddy** [comment]
hi @LevisNgigi , so upgrading to v5.0.0 is the only way to fix this issue or do we have any workaround or official fix for the v4.1.1? 
Thanks!

**Comment by lethehoa** [comment]
> > Got the problem **The query couldn't be loaded** in version 4.1.1, but I can't upgrade to the newest version now. Is there any way to fix this without upgrade?
> 
> Hey [@lethehoa](https://github.com/lethehoa) the fix is in v5.0.0 but I could help fix that for you if you do not want to upgrade.

Hi @LevisNgigi, Yeah of course, please help me

**Comment by SandadiLokanathaReddy** [comment]
> > > Got the problem **The query couldn't be loaded** in version 4.1.1, but I can't upgrade to the newest version now. Is there any way to fix this without upgrade?
> > 
> > 
> > Hey [@lethehoa](https://github.com/lethehoa) the fix is in v5.0.0 but I could help fix that for you if you do not want to upgrade.
> 
> Hi [@LevisNgigi](https://github.com/LevisNgigi), Yeah of course, please help me

Yes, any help in fixing this issue within v4.1.1 is much appreciated because v5.x is still an rc it seems, means not yet production ready.
